src/main/groovy/com/blackducksoftware/integration/hub/detect/Application.groovy

```diff
@@ -230,7 +231,7 @@ class Application {
                 logger.debug(e.getCause().getMessage(), e.getCause());
             }
             exitCodeType = e.getExitCodeType();
-        } else if (e instanceof IntegrationException) {
+        } else if (e instanceof IntegrationException || e instanceof IntegrationCertificateException) {
```

Comment:
Please, oh please tell me that IntegrationCertificateException isn't an IntegrationException...

Reply:
This actually should be IllegalStateException as the underlying code in HubServerConfigBuilder, line 72, can throw that.

Reply:
My original idea was to add IllegalStateException instead but then I figured many other processes could throw that. I wanted this specific handling when certificate is wrong or credentials were wrong, so I added the specific exception. Do you think it'd be a good idea to make IntegrationCertificateExcpetion an IntegrationException? I think that'd be the best option.